### PR TITLE
Update: Add Disabled attribute to CNSConfig which handles tenant state

### PR DESCRIPTION
**Pull request by ericrpowers**
#### Description
Add `Disabled` attribute to CNSConfig to handle tenant enablement. If this is set to true, the tenant is disabled or archived and to be deleted.

### CODE REVIEW

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "disable-tenant-sync",
    "component": "prismacloud",
    "pr-id": "83",
    "commit-sha": "29f9e3c3078ffa2d83d4ee05dfcc3e9471b8cea3",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "backend",
    "pr-id": "1658",
    "commit-sha": "8ae3bc987b392ed64365f4ddd19b2b199651b8eb",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "gaia",
    "pr-id": "791",
    "commit-sha": "88b31044597f81f1aa957c1384e9862585591a55",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "disable-tenant-sync",
    "component": "gaia",
    "pr-id": "791",
    "commit-sha": "88b31044597f81f1aa957c1384e9862585591a55",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "prismacloud",
    "pr-id": "83",
    "commit-sha": "29f9e3c3078ffa2d83d4ee05dfcc3e9471b8cea3",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "backend",
    "pr-id": "1658",
    "commit-sha": "261471cb553a473be00d0e216391d37c1705e411",
    "pipeline": "master"
  }
]
```

File: specs/cnsconfig.spec [hunk 1]
```diff
@@ -37,6 +37,13 @@ attributes:
     exposed: true
     stored: true
 
+  - name: enable
+    description: If `true` the tenant is enabled.
+    type: boolean
+    exposed: true
+    stored: true
+    default_value: true
```
Comment (t00f) [inline specs/cnsconfig.spec:45]: I know it sounds stupid but we always prefer using the default value of the boolean (right @primalmotion ? )

As by default a tenant should be enabled, I would call it disabled or archived. Default value would be false

**Review by t00f — COMMENTED** [review]
Let me read the rest of the project

File: specs/cnsconfig.spec (hunk 1 above)
Comment (ericrpowers) [reply specs/cnsconfig.spec:45]: I can use the abs `- '@disabled'`. Will swap over.

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "disable-tenant-sync",
    "component": "backend",
    "pr-id": "1658",
    "commit-sha": "cb74b0da1440a574b9fbcb2e88efc2787aaf92eb",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "gaia",
    "pr-id": "791",
    "commit-sha": "df6ec0323e2f57223415d573b27f7a1921d1fb45",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "prismacloud",
    "pr-id": "83",
    "commit-sha": "29f9e3c3078ffa2d83d4ee05dfcc3e9471b8cea3",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "disable-tenant-sync",
    "component": "gaia",
    "pr-id": "791",
    "commit-sha": "df6ec0323e2f57223415d573b27f7a1921d1fb45",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "prismacloud",
    "pr-id": "83",
    "commit-sha": "29f9e3c3078ffa2d83d4ee05dfcc3e9471b8cea3",
    "pipeline": "master"
  },
  {
    "project": "disable-tenant-sync",
    "component": "backend",
    "pr-id": "1658",
    "commit-sha": "ee529f26e7faca2aefec6f10387cfc046448c434",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]

**Pull request merged**